LICENSE.txt

```diff
@@ -1,21 +0,0 @@
-MIT License
-
-Copyright (c) 2020 Tomorrow
-
-Permission is hereby granted, free of charge, to any person obtaining a copy
-of this software and associated documentation files (the "Software"), to deal
-in the Software without restriction, including without limitation the rights
-to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-copies of the Software, and to permit persons to whom the Software is
-furnished to do so, subject to the following conditions:
-
-The above copyright notice and this permission notice shall be included in all
-copies or substantial portions of the Software.
```

Comment:
Not a lawyer, but although both the AGPL and MIT license allow the inclusion of MIT-licensed content in AGPL-licensed projects, I'm quite sure that doesn't mean (and in fact, the MIT license says it right here) that you are allowed to just delete the license text; rather, you should do something along the lines of:

> This project is licensed under the AGPL: (AGPL text)
>
>This project contains licensed under the MIT license: (MIT license text (including 'copyright year+author line))

Reply:
Thanks for bringing this to our attention @q-- 🙌 
We'll revisit this topic with our lawyers again and get back with an update here!

Reply:
Hi @q-- and thank you for flagging this issue!
We have indeed made a mistake when deleting the old license text and we will fix it as soon as possible. 

Regarding the difficulty of changing to a less restrictive license in the future, thank you again for flagging it! We are aware of it and it's something we have taken into consideration when choosing to change the license.

Reply:
The question is since we are using git the old license is still there if you check out a older commit than HEAD do we need to include the full license text or just mention that it contain MIT listened parts.

Reply:
> Hi @q-- and thank you for flagging this issue! We have indeed made a mistake when deleting the old license text and we will fix it as soon as possible.
> 
> Regarding the difficulty of changing to a less restrictive license in the future, thank you again for flagging it! We are aware of it and it's something we have taken into consideration when choosing to change the license.

Thanks for the quick response!

Now there's something else I realised a bit later, namely, that switching to the AGPL could also have consequences for the parts of the application code that you mean not to open-source.

From what I've gathered over the years it appears to me that _electricitymaps-contrib_ is part of a larger software product, and only the  _electricitymaps-contrib_ part is open source. That's not a problem on its own, but it could quickly become one when a certain set of conditions is met.

- Firstly,  _electricitymaps-contrib_ is now licensed under the _AGPL_, which requires 'Corresponding Source' to be published, among other scenario's, to users who are "interacting with [the Program] remotely through a computer network";*
 - (But note that, _as long as you are not accepting outside contributions after the license change_, the AGPL need not apply to the App.ElectrictyMap.org website, API or any other network resource that's using _electricitymaps-contrib_) because you can simply serve the existing parts created by outside contributors under the terms of the MIT license, which does not require the source code to be published. The parts created by your employees after that point shouldn't be a problem either, because (unless you messed up your employment contracts) copyright is owned by the company and, as the AGPL is a non-exclusive license, you can simply claim that ElectrictyMaps.org (or any other way of "interacting with [the Program] remotely through a computer network") does so under a different set of terms.)
- Secondly, outside contributions accepted _under the terms of the AGPL_; I'm quite sure this means that when those contributions by third parties become part of 'Program' and the 'Corresponding Source', you (that is, @electricitymaps) *are* required to adhere to the AGPL, including in the ways set forth in my first point; and
- Thirdly, and this is something I suspect but am not certain of is applicable, the proprietary part of your software is _interacting with_ electricitymaps-contrib _in a way that qualifies it to become part of the Product and Corresponding source.

In that scenario, the AGPL could  require you to open-source parts of the application you'd rather keep private.

But, after reading the part of the AGPL that defines 'Corresponding Source' (while working on this comment), I've become less certain. I haven't actually *seen* examples of such interaction occur, so I might just be worried about nothing.

Still, I'm bringing it up anyway in case it was overlooked. I've previously contributed to a project that went through a license change in a way that I'm quite certain was a violation of the GPL, so I might've become a bit too wary.

Reply:
> 

Thank you for your input on this matter. Indeed the reflection you make here is relevant and the concerns you are raising were also concerns we had when considering the change of license. After digging a bit more into the license terms with our lawyers we reached the conclusion that this potential problem you mention is not something we are subject to in our specific situation. 
Again thank you for worrying about this issue and for raising it with us in case we would have missed it, much appreciated!

Reply:
You're welcome. I'm glad to hear you've already taken this into consideration.

Reply:
> The question is since we are using git the old license is still there if you check out a older commit than HEAD do we need to include the full license text or just mention that it contain MIT listened parts.

I don't think so; "The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software."  suggests to me that it needs not only to be included when you do a `git clone` and checkout an old revision, but also when you download a ZIP
[from the project's home on GitHub](https://user-images.githubusercontent.com/2470175/216950236-b0921183-4cf2-49c9-adf4-5ee44aff5737.png) or the
![releases page](https://user-images.githubusercontent.com/2470175/216950522-4baedad2-4534-4438-9c00-704011b66d5d.png).

(As an aside, I suspect it "included in" also means it needs to be be present somewhere at ElectricityMaps.com as well. Perhaps it would be sufficient to only have it in the source code so people who look for it can find it. Or perhaps I'm wrong and the link to GitHub is sufficient, though that seems like stretching the definition of 'included' a bit too much to me.)